### ISSUE TITLE
vivado 2024.1

### ISSUE THREAD

**Issue by denruss** [RedPitaya/RedPitaya-FPGA#12]
C:\Projects\RedPitaya\FPGA>make project PRJ=v0.94 MODEL=Z10
vivado -source red_pitaya_vivado_project_Z10.tcl -tclargs v0.94 ""
process_begin: CreateProcess(NULL, vivado -source red_pitaya_vivado_project_Z10.tcl -tclargs v0.94 "", ...) failed.

**Comment by denruss** [comment]
Yes, I know that I need to use 2020.1, but it's too old, I don't want to keep a lot of virtual machines for different projects

**Comment by Lightsaver7** [comment]
I understand. As of now we have no plans to upgrade to a later version of Vivado. Perhaps it will happen in a year or two.
The base FPGA code is compatible with newer Vivado versions, the scripts for automatic project creation are the problem.

Here is a fix to make it work on newer Vivado versions if you really want to run it on newer versions:
https://redpitaya.readthedocs.io/en/latest/developerGuide/software/build/fpga/fpga.html#building-process

Check the red note in "point 7" of the "Building Process".

**Comment by peteasa** [comment]
With my Ubuntu 24.04 machine and I am now starting to use vivado 2023.2 for pyrpl fpga.  So if your requirement is for a more up to date vivado and you are happy to move away from the web based application then I may be able to help!

**Comment by peteasa** [comment]
Now using vivado 2024.2 for pyrpl fpga both built from the RedPitaya-FPGA source and the https://github.com/pyrpl-fpga/pyrpl/ source.
